model/spectrum.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * spectrum.h
+ */
+#ifndef DXTBX_MODEL_SPECTRUM_H
+#define DXTBX_MODEL_SPECTRUM_H
+
+#include <iostream>
+#include <cmath>
+#include <scitbx/vec3.h>
+#include <scitbx/array_family/shared.h>
+#include <scitbx/array_family/simple_io.h>
+#include <scitbx/array_family/simple_tiny_io.h>
+#include <dxtbx/error.h>
+#include "model_helpers.h"
+
+namespace dxtbx { namespace model {
+  typedef scitbx::af::shared<double> vecd;
+
+  /** A class to represent a 2D spectrum. */
+  class Spectrum {
+  public:
+    /** Default constructor: initialise all to zero */
+    Spectrum() {}
+
+    /**
+     * Initialise all the spectrum parameters.
+     * @param energies The spectrum energies (eV)
+     * @param energies The spectrum weights (unitless)
+     */
+    Spectrum(vecd energies, vecd weights)
+        : energies_(energies),
+          weights_(weights) {}
+
+    virtual ~Spectrum() {}
+
+    /* Get the spectrum energies (eV) */
+    vecd get_energies() const {
+      return energies_;
+    }
+
+    /* Get the spectrum weights (unitless) */
+    vecd get_weights() const {
+      return weights_;
+    }
+
+    double get_weighted_wavelength() const {
+      if (energies_.size() == 0)
+        return 0;
+      double weighted_sum = 0;
+      double summed_weights = 0;
+      for (size_t i = 0; i < energies_.size(); i++) {
+        weighted_sum += energies_[i] * weights_[i];
+        summed_weights += weights_[i];
+      }
+      DXTBX_ASSERT(weighted_sum > 0 && summed_weights > 0);
+      return 12398.4187 / (weighted_sum / summed_weights); //  eV per Å conversion factor
```

Comment:
Is this magic number worth defining somewhere central as a constant value? (see e.g. deg_as_rad etc.)

Reply:
I've used this factor hardcoded all over the place.  Would love it as a constant somewhere. Maybe a separate issue?

Reply:
Guess what? `scitbx::constants::factor_ev_angstrom` 🙂

Though this evaluates to a _slightly_ different constant value to your one - 

```
    //! Factor for keV <-> Angstrom conversion.
    /*!
      http://physics.nist.gov/PhysRefData/codata86/table2.html

      h = Plank's Constant = 6.6260755e-34 J s
      c = speed of light = 2.99792458e+8 m/s
      1 keV = 1.e+3 * 1.60217733e-19 J
      1 A = Angstrom = 1.e-10 m

      E = (h * c) / lamda;

      Exponents: (-34 + 8) - (3 - 19 - 10) = 0
     */
    static const double
    factor_kev_angstrom = 6.6260755 * 2.99792458 / 1.60217733;
    static const double
    factor_ev_angstrom  = 6626.0755 * 2.99792458 / 1.60217733;
```
=> 
```
>>> 6626.0755 * 2.99792458 / 1.60217733
12398.424468024265
```

Reply:
scitbx is out of date. [The Planck constant has changed.](https://web.archive.org/web/20181119214326/https://www.bipm.org/utils/common/pdf/CGPM-2018/26th-CGPM-Resolutions.pdf)

Reply:
➡️ https://github.com/cctbx/cctbx_project/pull/505

Reply:
The question was asked whether we can expose those numbers in Python. They are readily available (and maintained) in
```python
>>> import scipy.constants
>>> scipy.constants.Planck
6.62607015e-34
>>> scipy.constants.elementary_charge
1.602176634e-19
```
amongst many others.

Reply:
Is it the case that scipy is universally a dependency of dxtbx? If so, works for me.

Reply:
I don't think so, but we can just add it. [There is precedence in other packages.](https://github.com/cctbx/cctbx_project/blob/master/simtbx/nanoBragg/utils.py#L3)